### PR TITLE
fix: html fullscreen transitions stacking

### PR DESCRIPTION
**Pull request by codebytere**
Backport of #32905.

See that PR for details.

Notes: Fixes an issue with fullscreen transitions when HTML fullscreen is requested.

### CODE REVIEW

File: spec-main/chromium-spec.ts [hunk 1]
```diff
@@ -1677,8 +1677,37 @@ describe('iframe using HTML fullscreen API while window is OS-fullscreened', ()
...
+
+    await w.webContents.executeJavaScript(
+      "document.querySelector('iframe').contentWindow.postMessage('exitFullscreen', '*')"
+    );
+    await emittedOnce(w.webContents, 'leave-html-full-screen');
+
+    const width = await w.webContents.executeJavaScript(
+      "document.querySelector('iframe').offsetWidth"
+    );
+    expect(width).to.equal(0);
```
Comment (RaisinTen) [inline spec-main/chromium-spec.ts:1701]: This assertion is failing on macOS builds:
```console
not ok 1114 iframe using HTML fullscreen API while window is OS-fullscreened can fullscreen from out-of-process iframes (macOS)
  AssertionError: expected 1024 to equal 0
      at Context.<anonymous> (electron/spec-main/chromium-spec.ts:1701:22)
```

File: spec-main/chromium-spec.ts [hunk 2]
```diff
@@ -1677,8 +1677,37 @@ describe('iframe using HTML fullscreen API while window is OS-fullscreened', ()
...
+
+    w.setFullScreen(false);
+    await emittedOnce(w, 'leave-full-screen');
+  });
+
   // TODO(jkleinsc) fix this flaky test on WOA
   ifit(process.platform !== 'win32' || process.arch !== 'arm64')('can fullscreen from in-process iframes', async () => {
+    if (process.platform === 'darwin') await emittedOnce(w, 'enter-full-screen');
```
Comment (RaisinTen) [inline spec-main/chromium-spec.ts:1709]: I think this event is not emitted and so, this test times out on macOS:
```console
not ok 1115 iframe using HTML fullscreen API while window is OS-fullscreened can fullscreen from in-process iframes
  Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/distiller/project/src/electron/spec-main/chromium-spec.ts)
      at listOnTimeout (node:internal/timers:557:17)
      at processTimers (node:internal/timers:500:7)
```

**Review by RaisinTen — COMMENTED** [review]
There are some related macOS failures

**Comment by codebytere** [comment]
Gonna leave this in 19 and 20. Forgot this hinged on an upstream change not in 18.

**Pull request closed without merging**